### PR TITLE
L1T: fix duplicate dictionary problems in #33978

### PR DESCRIPTION
**Pull request by cecilecaillol**
#### PR description:

This is a fix addressing the comments from @makortel on https://github.com/cms-sw/cmssw/pull/33978

### CODE REVIEW

**Comment by cecilecaillol** [comment]
please test

File: L1Trigger/VertexFinder/src/classes.h [hunk 1]
```diff
@@ -13,44 +13,42 @@
...
+  edm::Wrapper<l1tVertexFinder::Stub> ws;
+  edm::Wrapper<edm::ValueMap<edm::Ptr<l1tVertexFinder::Stub>>> wvmptrstub;
+  edm::Wrapper<edm::ValueMap<edm::Ptr<l1tVertexFinder::TP>>> wvmprttp;
+  edm::Wrapper<edm::ValueMap<l1tVertexFinder::Stub>> wvmstub;
+  edm::Wrapper<edm::ValueMap<l1tVertexFinder::TP>> wvmtp;
+  edm::Wrapper<std::map<DetId, DetId>> wmdid;
+  edm::Wrapper<std::vector<l1tVertexFinder::Stub>> wvs;
+  edm::Wrapper<std::vector<l1tVertexFinder::TP>> wvtp;
+  edm::Wrapper<std::vector<l1tVertexFinder::Vertex>> wvvtx;
+  edm::Wrapper<l1tVertexFinder::InputData> wid;
```
Comment (makortel) [inline L1Trigger/VertexFinder/src/classes.h:51]: I mean that the contents of the struct (and the namespace)  can be removed.

File: L1Trigger/VertexFinder/src/classes_def.xml [hunk 2]
```diff
@@ -1,33 +1,28 @@
...
-  <class name="edm::Wrapper<edm::ValueMap<l1tVertexFinder::Stub> >"/>
-  <class name="edm::Wrapper<edm::ValueMap<l1tVertexFinder::TP> >"/>
-  <class name="edm::Wrapper<std::map<DetId,DetId>>"/>
-  <class name="edm::Wrapper<std::vector<l1tVertexFinder::Stub>>"/>
-  <class name="edm::Wrapper<std::vector<l1tVertexFinder::TP>>"/>
-  <class name="edm::Wrapper<std::vector<l1tVertexFinder::Vertex>>"/>
-  <class name="edm::Wrapper<l1tVertexFinder::InputData>"/>
-</lcgdict>
\ No newline at end of file
+  <class name="l1tVertexFinder::InputData" transient="true"/>
```
Comment (makortel) [inline L1Trigger/VertexFinder/src/classes_def.xml:2]: These should actually be 
```suggestion
  <class name="l1tVertexFinder::InputData" persistent="false"/>
```

File: SimDataFormats/TrackingAnalysis/src/classes_def.xml [hunk 3]
```diff
@@ -23,6 +23,8 @@
 
   <class name="TrackingParticleRefVector" />
   <class name="edm::Wrapper<TrackingParticleRefVector>"/>
+
+  <class name="std::map<edm::Ptr<TrackingParticle>,edm::RefToBase<TrackingParticle> >"/>
```
Comment (makortel) [inline SimDataFormats/TrackingAnalysis/src/classes_def.xml:27]: I'm still curious why `RefToBase<TrackingParticle>` is be needed. I don't see anything inheriting from `TrackingParticle`, so a polymorphic reference is not really needed (i.e. `edm::Ref` would suffice), and for any new use for polymorphic Refs the `edm::Ptr` is strongly preferred.

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

**Comment by civanch** [comment]
+1

**Comment by qliphy** [comment]
@cecilecaillol @makortel 
There are still some remaining Duplicate Dictionaries warnings:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cb8298/15911/dupDict/lostDefs.txt
./L1Trigger/VertexFinder/src/classes_def.xml
  Vertex : l1tVertexFinder::Vertex
  Vertex : vector< l1tVertexFinder::Vertex >
  Vertex : edm::Wrapper< l1tVertexFinder::Vertex >
  Vertex : edm::Wrapper< vector< l1tVertexFinder::Vertex > >

**Comment by makortel** [comment]
> Should we add
> {'VertexFinder' : ['l1tVertexFinder::Vertex']},
> in https://github.com/cms-sw/cmssw/blob/master/Utilities/ReleaseScripts/scripts/duplicateReflexLibrarySearch.py#L36

Yes, I suppose that would be the proper last step.

**Comment by qliphy** [comment]
@cecilecaillol  Would you please implement above change in this PR?  So we can test whether the issue can be fixed totally.  Thanks!

**Comment by qliphy** [comment]
please test

**Comment by cecilecaillol** [comment]
please test

File: Utilities/ReleaseScripts/scripts/duplicateReflexLibrarySearch.py [hunk 4]
```diff
@@ -42,6 +42,7 @@
          {'L1TrackTrigger'        : ['(TTStub|TTCluster|TTTrack).*Phase2TrackerDigi']},
          {'L1TCalorimeterPhase2'  : ['l1tp2::CaloTower.*']},
          {'L1TCalorimeter'        : ['l1t::CaloTower.*']},
+         {'VertexFinder' : ['l1tVertexFinder::Vertex']},
```
Comment (makortel) [inline Utilities/ReleaseScripts/scripts/duplicateReflexLibrarySearch.py:45]: Could you format the line similarly to surrounding?
```suggestion
         {'VertexFinder'         : ['l1tVertexFinder::Vertex']},
```
(I hope I got the number of spaces correctly)

**Comment by cecilecaillol** [comment]
please test

**Comment by smuzaffar** [comment]
+core
thanks @cecilecaillol  for the fix
FYI @mrodozov

**Comment by cecilecaillol** [comment]
+l1

**Comment by civanch** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**